### PR TITLE
improvement: Consolidated wallet syncing

### PR DESCRIPTION
**Pull request by OAuthBringer**
Re-enables bitflyer refresh and consolidates logic into a single implementation.

### CODE REVIEW

File: app/models/gemini_connection.rb [hunk 1]
```diff
@@ -64,27 +64,30 @@ def fetch_refresh_token
...
+      user = users.find { |u| u.is_verified && u.status == "Active" }
 
-    users = Gemini::Account.find(token: access_token).users
-    user = users.find { |u| u.is_verified && u.status == "Active" }
+      # If we couldn't find a verified account we'll take the first user.
+      user ||= users.first
 
-    # If we couldn't find a verified account we'll take the first user.
-    user ||= users.first
+      update(
```
Comment (OAuthBringer) [inline app/models/gemini_connection.rb:77]: @tsmartt DO we really want to run this every time the user logs in?  I understand the refresh_authorization! piece but syncing over their data on every login?

Comment (tsmartt) [reply app/models/gemini_connection.rb:77]: Once a month should be fine, in our job

File: test/jobs/sync/wallet_connection_job_test.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  describe "#perform" do
+    describe "without args" do
+      test "it should return an integer" do
+        result = Oauth2BatchRefreshJob.perform_now
+        assert result > limit
+      end
+    end
+
+    describe "witht args" do
```
[the remark this answers is not in the document]
Comment (OAuthBringer) [reply test/jobs/sync/wallet_connection_job_test.rb]: I mean do you want me to rerun the whole build because of a typo?

**Review by tsmartt — APPROVED** [review]
Approving due to need, please fix typo in followup

**Pull request merged**